### PR TITLE
Add final REST interface for QueueManager

### PR DESCRIPTION
**Pull request by Lnaden**
## Description
Similar to #130 and #142, this PR adds the final REST interfaces for the API

I believe this is the last REST validation we need to add for now to have them all in, even if rigorous validation is not fully preset beyond things like `Dict[str, Any]`

Advances #16 

## Status
- [x] Ready to go

### CODE REVIEW

**Comment by Lnaden** [comment]
I'm adding some comments here for an issue which came up in case we run into it again and forget why.

tl;dr: Always use `Dict[Any, Any]` at minimum instead of `Dict` when typing with Pydantic

If you make a Pydantic object with the `typing.Dict` type and don't include the `[..., ...]` args, it will throw the following pair of errors:

```
TypeError: issubclass() arg 1 must be a class

The above exception was the direct cause of the following exception:
...
RuntimeError: error checking inheritance of ~KT (type: KT)
```

This is NOT the Boltzmann Temperature (kT), but instead the `typing` module's generic KeyType and ValueType objects which it uses as placeholders. Pydantic has no concept of these as doing `Dict` without `Dict[]` only throws this error in Python 3.7. (as of Pydantic 0.18)

For example, assuming Pydantic 0.18, the below will work in its entirety Python 3.6, but in Python 3.7, class `X` and class `Y` will.

```python
from pydantic import BaseModel
from typing import Dict, Any

class X(BaseModel):
    x: Dict

class Y(BaseModel):
    y: Dict[str, Any]
```

**Comment by Lnaden** [comment]
The py37 tests do pass now? Unless you meant "must pass?"
